package.json

```diff
@@ -136,19 +148,43 @@
                 "command": "lean.displayGoal",
                 "key": "ctrl+shift+enter",
                 "mac": "cmd+shift+enter",
-                "when": "editorTextFocus"
+                "when": "editorTextFocus && editorLangId == lean"
             },
             {
                 "command": "lean.batchExecute",
                 "key": "ctrl+shift+r",
                 "mac": "cmd+shift+r",
-                "when": "editorTextFocus"
+                "when": "editorTextFocus && editorLangId == lean"
             }
-        ]
+        ],
+        "menus": {
+            "commandPalette": [
```

Comment:
Yes, `commandPalette` (`ctrl+shift+p`) is usually filled up with contributions in `commands`. By adding our own entries to `commandPalette` we can overwrite the `when` attribute. So we can control that some commands are only shown in the command palette when we are in a Lean editor. So when you are in a normal `*.txt` file, `lean.showGoals` should not be available in `ctrl+shift+p`.